public/app/features/dashboard/components/Inspector/InspectDataTab.tsx

```diff
@@ -28,8 +46,44 @@ export class InspectDataTab extends PureComponent<Props> {
     saveAs(blob, dataFrame.name + '-' + new Date().getUTCDate() + '.csv');
   };
 
+  onSelectedFrameChanged = (item: SelectableValue<number>) => {
+    this.setState({ dataFrameIndex: item.value || 0 });
+  };
+
+  onTransformationChange = (value: SelectableValue<DataTransformerID>) => {
+    this.setState({ transformId: value.value });
+  };
+
+  getTransformedData(): DataFrame[] {
+    const { transformId, transformationOptions } = this.state;
+    const { data } = this.props;
+
+    if (!data) {
+      return [];
+    }
+
+    const currentTransform = transformationOptions.find(item => item.value === transformId);
+
+    if (currentTransform && currentTransform.transformer.id !== DataTransformerID.noop) {
+      return transformDataFrame([currentTransform.transformer], data);
+    }
+    return data;
+  }
+
+  getProcessedData(): DataFrame[] {
+    return applyFieldOverrides({
+      data: this.getTransformedData(),
+      theme: config.theme,
```

Comment:
Out of the scope for this PR but passing theme to field overrides seems weird to me.

Reply:
It's used to derive threshold colors from the palette

Reply:
I have a similar problem with this, but haven't found a better way:(